enhancements/single-node/relocatable-single-node-openshift.md

```diff
@@ -0,0 +1,410 @@
+Upon deployment at the edge site the reconfiguration service will read the site-specific configuration (relocation-config) and reconfigure the following:
+(see [Alternatives for SNO reconfiguration](#Alternatives for SNO reconfiguration).
+
+1. IP address and DNS server
+The relocation config may contain the desired network configuration for the edge site.
+The re-configuration service will expect network overrides as [nmstate](https://github.com/nmstate/nmstate) nmconnection files.
+The reconfiguration service will apply the nmstate configs by copying them to `/etc/NetworkManager/system-connections/` and restart `NetworkManager` as the first reconfiguration step.
+Once the networking is configured, the reconfiguration service will start kubelet and wait for the kube-apiserver to be available.
+
+2. Configuring API resources:
+Once the kube-apiserver is available the reconfiguration service will create a cluster-relocation CR with all the site specific configuration for the edge site.
+
+5. cluster Domain
+While changing the default domain of OCP is not currently supported, we can [modify the domain](https://docs.openshift.com/container-platform/4.12/networking/ingress-operator.html) for all related components. To achieve this, several tasks are performed by the re-configuration service:
+a. Create new certificates for the console and authentication with the new domain suffix.
+b. Add componentRoutes for the console and authentication.
+c. Add namedCert into the apiserver
+d. Add appsDomain to use the new domain suffix.
+e. Replace all existing routes, ensuring that the components of the cluster utilize the updated domain suffix. 
+
+3. Reconfigure the pull secret
+Changing the pull secret is supported by OCP. In case the relocation-config contains pull secret, the re-configuration service will execute `oc set data secret/pull-secret -n openshift-config` with the new pull secret as described [here](https://access.redhat.com/solutions/4902871).
+4. Reconfigure the ImageContentSourcePolicy
+In case the relocation-config contains ImageContentSourcePolicy the re-configuration service will create a new ImageContentSourcePolicy resource with the updated policy.
+
+5. Add new trusted CA for a mirror registry.
+
+6. Hostname
+Changing the node hostname is not currently supported by OpenShift.
+We need to investigate the impact of this change on OpenShift.
+
+7. Cluster name
+Changing the cluster name is not currently supported by OpenShift.
+We need to investigate the impact of this change on OpenShift.
+
+#### Certificate expiry
+
+Upon deployment at the edge site, the kubelet apiserver client and kubelet serving certificates might be expired, since the initial certificate is currently valid for 24 hours.
+In such case the new systemd service will auto approve the necessary kubelet CSRs until valid certificate is issued for kubelet to authenticate with the kube-apiserver and for the node to register.
```

Comment:
there are actually many more that would expire and need rotation:
https://issues.redhat.com/browse/OCPSTRAT-714

Here's the epic for etcd: https://issues.redhat.com/browse/ETCD-445